shark/shark_importer.py

```diff
@@ -51,60 +52,52 @@ def __init__(
             )
             return
 
-        if self.model_source_hub == "tfhub":
-            # compile and run tfhub tflite
-            if self.model_type == "tflite":
-                load_model_success = self.load_tflite_model()
-                if load_model_success == False:
-                    print("Error, load tflite model fail")
-                    return
-
-                if (self.input_details == None) or (
-                    self.output_details == None
-                ):
-                    print(
-                        "Setting up tflite interpreter to get model input details"
-                    )
-                    self.tflite_interpreter = tf.lite.Interpreter(
-                        model_path=self.tflite_saving_file
-                    )
-                    self.tflite_interpreter.allocate_tensors()
-                    # default input initialization
-                    (
-                        self.input_details,
-                        self.output_details,
-                    ) = self.get_model_details()
-                    inputs = self.generate_inputs(
-                        self.input_details
-                    )  # device_inputs
-                self.setup_inputs(inputs)
+        print("Setting up for TMP_WORK_DIR")
+        self.workdir = os.path.join(
+            os.path.dirname(__file__), "./../gen_shark_tank"
+        )
+        os.makedirs(self.workdir, exist_ok=True)
+        print(f"TMP_WORK_DIR = {self.workdir}")
+
+        # compile and run tfhub tflite
+        if self.model_type == "tflite":
+            load_model_success = self.load_tflite_model()
+            if not load_model_success:
+                print("Error, load tflite model fail")
+                return
+
+            if (self.input_details is None) or (self.output_details is None):
+                print(
+                    "Setting up tflite interpreter to get model input details"
+                )
+                self.setup_interpreter()
+
+                inputs = self.generate_inputs(
+                    self.input_details
+                )  # device_inputs
+            self.setup_inputs(inputs)
+
+        elif self.model_type in ["tensorflow, tf, torch, pytorch"]:
+            print(self.model_type, " Not Implemented yet")
 
     def load_tflite_model(self):
```

Comment:
All the tf_lite related functions should go into tf_lite_utils.py. This will make this class clean. So you just have to call get_tf_lite_mlir(model_name, url ...) and it will get the model with all the metadata.

Reply:
Done